src/app/workflow/workflow-file-editor/workflow-file-editor.component.ts

```diff
@@ -77,23 +77,29 @@ export class WorkflowFileEditorComponent extends FileEditing {
    * Creates a new version based on changes made
    */
   saveVersion() {
-    const message = 'Save Version';
     const combinedSourceFiles = this.getCombinedSourceFiles();
     const newSourceFiles = this.commonSaveVersion(this.originalSourceFiles, combinedSourceFiles);
     this.alertService.start('Updating hosted workflow');
     this.hostedService.editHostedWorkflow(
-        this.id,
-        newSourceFiles).subscribe((workflow: Workflow) => {
-          this.toggleEdit();
+      this.id,
+      newSourceFiles).subscribe((workflow: Workflow) => {
+        this.toggleEdit();
+        if (workflow) {
           this.workflowsService.getWorkflow(workflow.id).subscribe((workflow2: Workflow) => {
             this.alertService.detailedSuccess();
             this.workflowService.setWorkflow(workflow2);
-          });
-        }, error =>  {
-          if (error) {
+          }, error => {
             this.alertService.detailedError(error);
-          }
+          });
+        } else {
+          // Probably encountered a 204
```

Comment:
The downside of using a generated client library. :) Good thing this method never returns 201; how would you ever distinguish a 201 and 204?

Reply:
there's an observe parameter in the function that you can specify `response` instead of the default which is `body`.   Once you have the response, you can figure out what the status code was.  I personally thought i was overkill for this, so I just left it as is.

Reply:
👍 Did not know that. I was frustrated by the inability to write the API I wanted for POSTing zips because of client code generation and vented incorrectly here.

Reply:
👍 Did not know that. I was frustrated by the inability to write the API I wanted for POSTing zips because of client code generation and vented incorrectly here.